code/src/App.js

```diff
@@ -1,9 +1,83 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
+
+import { HAPPY_THOUGHTS_URL, HEARTS_URL } from './reusable/urls';
+
+import MessageForm from 'components/MessageForm';
+import MessageList from 'components/MessageList';
 
 export const App = () => {
+  const [messageList, setMessageList] = useState([]);
+  const [messageNew, setMessageNew] = useState(['']);
+
+  useEffect(() => {
+    fetchMessageList();
+  }, []);
+
+  const fetchMessageList = () => {
+    fetch(HAPPY_THOUGHTS_URL)
+      .then(res => res.json())
+      .then(inputs => setMessageList(inputs))
+      .catch(err => console.error(err));
+  }
+
+  const handleMessageNewChange = (e) => {
+    setMessageNew(e.target.value);
+  }
+
+  const handleFormSubmit = (e) => {
+    e.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: messageNew })
+    }
+
+    fetch(HAPPY_THOUGHTS_URL, options)
+      .then(res => res.json())
+      .then(recievedMessage => setMessageList([...messageList, recievedMessage]))
+      .catch(err => console.error(err));
```

Comment:
I saw that you mentioned that your newest message was sent to the bottom. I did this trick that i read about in the slack channel where you instead of having it in the order you have now `[...messageList, recievedMessage]` you can reverse the order by putting recievedMessage before the spread operator like this: [recievedMessage, ...messageList]. I actually dont know if this is the legit way of doing but you could try it out 😸